### PR TITLE
Speed up adaptive threshold in findChessboardCorners

### PR DESCRIPTION
**Pull request by MaximSmolskiy**
### Pull Request Readiness Checklist

If `block_size` hasn't been changed between iterations for same `k`, then all `adaptiveThreshold` arguments will be same and we can reuse result from previous iteration.

I tested this PR with benchmark
```
python3 objdetect_benchmark.py --configuration=generate_run --board_x=7 --path=res_chessboard --synthetic_object=chessboard
```
PR speed up chessboards detection by `7.5/17%` without any changes in detected chessboards number:
```
cell_img_size = 100 (default)

before
                                 category  detected chessboard  total detected chessboard  total chessboard  average detected error chessboard
                                      all             0.904167                      13020             14400                           0.600512
Total detected time:  107.27875600000003 sec

after
                                 category  detected chessboard  total detected chessboard  total chessboard  average detected error chessboard
                                      all             0.904167                      13020             14400                           0.600512
Total detected time:  99.0223499999999 sec

----------------------------------------------------------------------------------------------------------------------------------------------

cell_img_size = 10

before
                                 category  detected chessboard  total detected chessboard  total chessboard  average detected error chessboard
                                      all             0.539792                       7773             14400                           4.209964
Total detected time:  2.989205999999999 sec

after
                                 category  detected chessboard  total detected chessboard  total chessboard  average detected error chessboard
                                      all             0.539792                       7773             14400                           4.209964
Total detected time:  2.4802350000000013 sec
```

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Review by asmorkalov — APPROVED** [review]

File: modules/calib3d/src/calibinit.cpp [hunk 1]
```diff
@@ -595,9 +597,19 @@ bool findChessboardCorners(InputArray image_, Size pattern_size,
...
+                    {
+                        adaptiveThreshold( img, thresh_img, 255, ADAPTIVE_THRESH_MEAN_C, THRESH_BINARY, block_size, (k/2)*5 );
+                        dilate( thresh_img, thresh_img, Mat(), Point(-1, -1), dilations );
+                        thresh_img.copyTo(prev_thresh_img);
+                    }
+                    else
+                    {
+                        if (dilations > 0)
+                            dilate( prev_thresh_img, prev_thresh_img, Mat(), Point(-1, -1), 1 );
+                        prev_thresh_img.copyTo(thresh_img);
```
Comment (AleksandrPanov) [inline modules/calib3d/src/calibinit.cpp]:
```
                    else if (dilations > 0)
                    {
                        dilate( prev_thresh_img, prev_thresh_img, Mat(), Point(-1, -1), 1 );
                        prev_thresh_img.copyTo(thresh_img);
                    }
```

**Review by AleksandrPanov — APPROVED** [review]
👍

**Comment by MaximSmolskiy** [comment]
@AleksandrPanov Moving `int prev_block_size = -1;` outside from loop on variable `k` should not be equivalent change. Variable `k` is used as argument in `adaptiveThreshold` function, so we shouldn't reuse result from different `k` - only for same `k`. This change decreased detected chessboards number
```
cell_img_size = 100 (default)

                                 category  detected chessboard  total detected chessboard  total chessboard  average detected error chessboard
                                      all             0.469306                       6758             14400                           4.178116

----------------------------------------------------------------------------------------------------------------------------------------------

cell_img_size = 10

                                 category  detected chessboard  total detected chessboard  total chessboard  average detected error chessboard
                                      all             0.895833                      12900             14400                           0.600768
```

**Comment by AleksandrPanov** [comment]
> @AleksandrPanov Moving `int prev_block_size = -1;` outside from loop on variable `k` should not be equivalent change. Variable `k` is used as argument in `adaptiveThreshold` function, so we shouldn't reuse result from different `k` - only for same `k`. This change decreased detected chessboards number
> 
> ```
> cell_img_size = 100 (default)
> 
>                                  category  detected chessboard  total detected chessboard  total chessboard  average detected error chessboard
>                                       all             0.469306                       6758             14400                           4.178116
> 
> ----------------------------------------------------------------------------------------------------------------------------------------------
> 
> cell_img_size = 10
> 
>                                  category  detected chessboard  total detected chessboard  total chessboard  average detected error chessboard
>                                       all             0.895833                      12900             14400                           0.600768
> ```

@MaximSmolskiy, thx, fixed

**Comment by AleksandrPanov** [comment]
Now the results are correct:
`--configuration=generate_run --board_x=7 --path=res_chessboard --synthetic_object=chessboard`
image

`--configuration=generate_run --board_x=7 --path=res_chessboard --synthetic_object=chessboard --cell_img_size=10`
image

**Pull request merged**